.upstream-tests/test/cuda/atomics/atomic.ext/atomic_fetch_max.pass.cpp

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// UNSUPPORTED: libcpp-has-no-threads, pre-sm-60
+// UNSUPPORTED: windows && pre-sm-70
+
+// <cuda/std/atomic>
+
+#include <cuda/std/atomic>
+#include <cuda/std/type_traits>
+#include <cuda/std/cassert>
+
+#include "test_macros.h"
+#include "atomic_helpers.h"
+#include "cuda_space_selector.h"
+
+template <class T, template<typename, typename> typename Selector, cuda::thread_scope>
+struct TestFn {
+  __host__ __device__
+  void operator()() const {
+    // Test greater
+    {
+        typedef cuda::atomic<T> A;
+        Selector<A, constructor_initializer> sel;
+        A & t = *sel.construct();
+        t = T(1);
+        assert(t.fetch_max(2) == T(1));
```

Comment:
Is this expected? I believe the tests should operate similarly to `fetch_add` where the result is the current value of the atomic.

Reply:
This looks right to me.

Reply:
If that's the case then the documentation and example is wrong.

https://nvidia.github.io/libcudacxx/extended_api/synchronization_primitives/atomic/fetch_max.html

Reply:
The docs and the example suggest that this isn't an RMW operation, but then this wouldn't be very useful. The example is also clearly not how `atom.{min,max}` work in PTX.

Reply:
I hadn't dug deep enough on this. I bypassed the PTX enhancements. Will append another commit to fix.